controllers/machineset_controller.go

```diff
@@ -226,6 +227,29 @@ func (r *MachineSetReconciler) reconcile(ctx context.Context, cluster *clusterv1
 		filteredMachines = append(filteredMachines, machine)
 	}
 
+	var errs []error
+	for _, machine := range filteredMachines {
+		if conditions.IsFalse(machine, clusterv1.ConditionOwnerRemediated) {
+			logger.Info("Deleting unhealthy machine", "machine", machine.GetName())
+			patch := client.MergeFrom(machine.DeepCopy())
+			if err := r.Client.Delete(ctx, machine); err != nil {
+				errs = append(errs, errors.Wrap(err, "failed to delete"))
+				conditions.MarkFalse(machine, clusterv1.ConditionOwnerRemediated, "FailedToDelete", clusterv1.ConditionSeverityError, "error: %v", err)
```

Comment:
I'm not sure that it is correct to set an error condition on the node if the machine set fails to delete...
(plus, use a constant for condition reason)

Reply:
Yeah, I don't know why this would ever happen, but the idea of this is that this condition is already set, we are just increasing the severity and changing the reason. We don't have to do this though.